source/server/http/admin.cc

```diff
@@ -664,8 +664,9 @@ void AdminFilter::onComplete() {
   Http::HeaderMapPtr header_map{new Http::HeaderMapImpl};
   RELEASE_ASSERT(request_headers_);
 
-  AdminStreamImpl admin_stream(*callbacks_, *request_headers_, on_destroy_callbacks_);
-  Http::Code code = parent_.runCallback(path, *header_map, response, admin_stream);
+  std::unique_ptr<AdminStreamImpl> admin_stream =
+      std::make_unique<AdminStreamImpl>(*callbacks_, *request_headers_, on_destroy_callbacks_);
+  Http::Code code = parent_.runCallback(path, *header_map, response, *admin_stream);
```

Comment:
I think my suggestion was not clear.  I was suggesting you make this unique_ptr<AdminStreamImpl> be a member variable of AdminFilter, and should own the headers, callbacks, and destroy-callbacks exclusively.

The existing of this as an independent class referencing member variables of AdminStream seemed perilous and I think that change makes the ownership semantics much clearer and easy to reason about, without affecting the actual flow of the logic.

WDYT?

Reply:
I don't mind this change.  But need to check if callbacks can be on a unique pointer,  since it points to outside admin.